luarules/gadgets/unit_dgun_behaviour.lua

```diff
@@ -79,13 +80,18 @@ end
 
 function gadget:UnitPreDamaged(unitID, unitDefID, unitTeam, damage, paralyzer, weaponDefID, projectileID, attackerID, attackerDefID, attackerTeam)
 	if dgunDamages[weaponDefID] and isCommander[unitDefID] and isCommander[attackerDefID] then
-		damagedUnits[projectileID] = damagedUnits[projectileID] or {}
-		if damagedUnits[projectileID][unitID] then
-			return 0
-		else
-			damagedUnits[projectileID][unitID] = true
-			return dgunDamages[weaponDefID][unitArmorType[unitDefID]]
-		end
+--		damagedUnits[projectileID] = damagedUnits[projectileID] or {}
+	--	if damagedUnits[projectileID][unitID] then
+	--		return 0
+--		else
+	--		damagedUnits[projectileID][unitID] = true
+	--		return dgunDamages[weaponDefID][unitArmorType[unitDefID]]
+--		end
+--	if isDGun[weaponDefID] and isCommander[unitDefID] and isCommander[attackerDefID] then
+		Spring.DeleteProjectile(projectileID)
+		local XPos, YPos, ZPos = Spring.GetUnitPosition(unitID)		
+		Spring.SpawnCEG("commander-spawn-explo", XPos, YPos, ZPos,0,0,0,0,0)
 	end
-	return damage
+	--end
+	--return damage
```

Comment:
```suggestion
	return damage
```
In the case of not a Commander getting hit with a d-gun, the function still needs to return the damage to apply.